Data-Structures/src/main/java/datastructures/tree/BinarySearchTree.java

```diff
@@ -0,0 +1,63 @@
+package datastructures.tree;
+
+public class BinarySearchTree {
+    Node root;
+    //    Node runner;
+    public void addNode(int value){
+        Node newNode = new Node (value);  //create new node with value and name
+        if(root == null){     //check if root exists. if it doesn't, new node becomes the root.
+            root = newNode;
+        }
+        else{       //if root already exists, make root the pivot point of reference for left and right nodes as well as parent.
+            Node runner = root;
+            Node parent;
+            while (1 < 2) {
```

Comment:
Really try to avoid these infinite loops... find some conditional logic to put here based on your algorithm. Also, I would highly recommend to do this method using recursion. This is where recursion is a great choice for your algorithm, and also a great learning opportunity. 👍